test/sparse/sparse.jl

```diff
@@ -1672,8 +1672,8 @@ end
 
 @testset "sparse matrix cond" begin
     local A = sparse(reshape([1.0], 1, 1))
-    Ac = sprandn(20, 20,.5) + im* sprandn(20, 20,.5)
-    Ar = sprandn(20, 20,.5)
+    Ac = Diagonal(fill(1., 20)) + (sprandn(20, 20,.5) + im*sprandn(20, 20,.5))/40
```

Comment:
This will probably make it too well condition. I did some simulations locally and
```
Ar = sprandn(20, 20,.5) + eps()*I
```
seems sufficient to avoid the singular case while still allowing for some variation in the condition number. It is not necessary to adjust the complex version since adding the two random patterns makes it extremely unlikely that the matrix is singular.

Reply:
This solution also seems to take care of the issue reported in https://github.com/JuliaLang/julia/pull/25218#issuecomment-353253004

Reply:
Revised accordingly to worsen conditioning. (Not something I anticipated ever saying 😉.)